tests/e2e/system_test/test_mcg_replication_with_disruptions.py

```diff
@@ -0,0 +1,241 @@
+            mcg_obj=mcg_obj_session,
+            amount=5,
+            pattern="first-write-",
+        )
+        logger.info(f"Written objects: {written_random_objects}")
+
+        compare_bucket_object_list(
+            mcg_obj_session, source_bucket_name, target_bucket_name
+        )
+        logger.info("Uni-directional bucket replication working as expected")
+
+        # change from uni-directional to bi-directional replication policy
+        bi_replication_policy_dict = {
+            "spec": {
+                "additionalConfig": {
+                    "replicationPolicy": json.dumps(
+                        [
+                            {
+                                "rule_id": "basic-replication-rule-2",
+                                "destination_bucket": source_bucket_name,
+                            }
+                        ]
+                    )
+                }
+            }
+        }
+        OCP(
+            namespace=config.ENV_DATA["cluster_namespace"],
+            kind="obc",
+            resource_name=target_bucket_name,
+        ).patch(params=json.dumps(bi_replication_policy_dict), format_type="merge")
+        logger.info(
+            "Patch ran successfully! Changed the replication policy from uni to bi directional"
+        )
+
+        # write objects to the second bucket and see if it's replicated on the other
+        written_random_objects = write_random_test_objects_to_bucket(
+            awscli_pod_session,
+            target_bucket_name,
+            test_directory_setup.origin_dir,
+            mcg_obj=mcg_obj_session,
+            amount=3,
+            pattern="second-write-",
+        )
+        logger.info(f"Written objects: {written_random_objects}")
+        compare_bucket_object_list(
+            mcg_obj_session, source_bucket_name, target_bucket_name
+        )
+        logger.info("Bi directional bucket replication working as expected")
+
+        # write some object to any of the bucket, followed by immediate cluster restart
+        written_random_objects = write_random_test_objects_to_bucket(
+            awscli_pod_session,
+            target_bucket_name,
+            test_directory_setup.origin_dir,
+            mcg_obj=mcg_obj_session,
+            amount=1,
+            pattern="third-write-",
+        )
+        logger.info(f"Written objects: {written_random_objects}")
+
+        node_list = get_worker_nodes()
+        node_objs = get_node_objs(node_list)
+        nodes.restart_nodes(node_objs, force=False, timeout=500)
+        retry(
+            (CommandFailed, TimeoutError, AssertionError, ResourceWrongStatusException),
+            tries=60,
+            delay=15,
+        )(ocp.wait_for_cluster_connectivity(tries=400))
+        retry(
+            (CommandFailed, TimeoutError, AssertionError, ResourceWrongStatusException),
+            tries=60,
+            delay=15,
+        )(wait_for_nodes_status(timeout=1800))
```

Comment:
Do we need retries for wait_for_nodes_status(timeout=1800) and wait_for_cluster_connectivity(tries=400) ? Can't the respective function timeouts/tries enough?

Reply:
This is for "just in case" scenario :smile: but I dont think its needed